src/explore-education-statistics-admin/src/pages/release/edit-release/content/helpers.ts

```diff
@@ -0,0 +1,345 @@
+import {
+  EditableContentBlock,
+  ExtendedComment,
+} from '@admin/services/publicationService';
+import { releaseContentService } from '@admin/services/release/edit-release/content/service';
+import {
+  ContentBlockAttachRequest,
+  ContentBlockPostModel,
+} from '@admin/services/release/edit-release/content/types';
+import { Dictionary } from '@admin/types';
+import {
+  AbstractRelease,
+  ContentSection,
+} from '@common/services/publicationService';
+import { AxiosResponse } from 'axios';
+import { RemoveBlockFromSection } from './actions';
+import { Dispatch } from './ReleaseContext';
+
+const contentSectionComments = (
+  contentSection?: ContentSection<EditableContentBlock>,
+) => {
+  if (
+    contentSection &&
+    contentSection.content &&
+    contentSection.content.length > 0
+  ) {
+    return contentSection.content.reduce<ExtendedComment[]>(
+      (allCommentsForSection, content) =>
+        content.comments
+          ? [...allCommentsForSection, ...content.comments]
+          : allCommentsForSection,
+      [],
+    );
+  }
+
+  return [];
+};
+
+const getUnresolveComments = (release: AbstractRelease<EditableContentBlock>) =>
+  [
+    ...contentSectionComments(release.summarySection),
+    ...contentSectionComments(release.keyStatisticsSection),
+    ...release.content
+      .filter(_ => _.content !== undefined)
+      .reduce<ExtendedComment[]>(
+        (allComments, contentSection) => [
+          ...allComments,
+          ...contentSectionComments(contentSection),
+        ],
+        [],
+      ),
+  ].filter(comment => comment !== undefined && comment.state === 'open');
+
+export async function getReleaseContent(
+  dispatch: Dispatch,
+  releaseId: string,
+  errorHandler: (err: AxiosResponse) => void,
+) {
+  dispatch({ type: 'CLEAR_STATE' });
+  try {
+    const {
+      release,
+      availableDataBlocks,
+    } = await releaseContentService.getContent(releaseId);
+    const canUpdateRelease = await permissionsService.canUpdateRelease(
+      releaseId,
+    );
+    dispatch({
+      type: 'SET_STATE',
+      payload: {
+        unresolvedComments: getUnresolveComments(release),
+        release,
+        availableDataBlocks,
+        canUpdateRelease,
+      },
+    });
+  } catch (err) {
+    errorHandler(err);
+  }
+}
+
+export async function updateAvailableDataBlocks(
+  dispatch: Dispatch,
+  releaseId: string,
+  errorHandler: (err: AxiosResponse) => void,
+) {
+  try {
+    const availableDataBlocks = await releaseContentService.getAvailableDataBlocks(
+      releaseId,
+    );
+    dispatch({
+      type: 'SET_AVAILABLE_DATABLOCKS',
+      payload: { availableDataBlocks },
+    });
+  } catch (err) {
+    errorHandler(err);
+  }
+}
+
+export async function deleteContentSectionBlock(
+  dispatch: Dispatch,
+  releaseId: string,
+  sectionId: string,
+  blockId: string,
+  sectionKey: RemoveBlockFromSection['payload']['meta']['sectionKey'],
```

Comment:
Here's the nasty `RemoveBlockFromSection['payload']['meta']` type rearing its head. Use the refactored meta type instead (same for rest of the file).

Reply:
done replaced `RemoveBlockFromSection['payload']['meta']['sectionKey']` with new type `ContentSectionKeys`